x/nft/msg.go

```diff
@@ -0,0 +1,56 @@
+package nft
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+var _ weave.Msg = (*AddApprovalMsg)(nil)
+var _ weave.Msg = (*RemoveApprovalMsg)(nil)
+
+const (
+	pathAddApproval    = "nft/add-approval"
+	pathRemoveApproval = "nft/remove-approval"
+)
+
+// Path returns the routing path for this message
+func (*AddApprovalMsg) Path() string {
+	return pathAddApproval
+}
+
+// Path returns the routing path for this message
+func (*RemoveApprovalMsg) Path() string {
+	return pathRemoveApproval
+}
+
+func (m *AddApprovalMsg) Validate() error {
+	if err := weave.Address(m.Address).Validate(); err != nil {
+		return err
+	}
+	if m.Action == "" {
```

Comment:
All actions are defined internally. let's have some convention here so that we can validate them via regexp. like `^[a-z_]{4,32}$` WDYT?

Reply:
I think a-zA-Z will be better, otherwise looks good to me. it's just a bit more readable camel-cased I'd say. Not a fan of _, but if you want to use it - then we can go for upper-case only java-style constant names like so ACTION_NAME.
then it's `^[A-Z_]{4,32}$`

Reply:
Actually yes, the `ACTION_NAME` format is readable and suitable for constants, like actions.